spec/components/schools/mentors/summary_component_spec.rb

```diff
@@ -0,0 +1,70 @@
+RSpec.describe Schools::Mentors::SummaryComponent, type: :component do
+  let(:school) { create(:school) }
+  let(:mentor_teacher) { create(:teacher, trs_first_name: 'Naruto', trs_last_name: 'Uzumaki') }
+  let(:started_on) { Date.new(2023, 9, 1) }
+  let(:mentor) { create(:mentor_at_school_period, teacher: mentor_teacher, school:, started_on:, finished_on: nil) }
+
+  context 'with no ECTs' do
+    it 'shows No ECTs assigned' do
+      render_inline(described_class.new(mentor:, school:))
+      expect(rendered_content).to have_css('.govuk-summary-list__row', text: 'Assigned ECTs')
+      expect(rendered_content).to have_css('.govuk-summary-list__value', text: 'No ECTs assigned')
+    end
+  end
+
+  context 'with less than or equal to 5 ECTs' do
+    let!(:ects) do
+      (1..5).map do |i|
+        teacher = create(:teacher, trs_first_name: "First name #{i}", trs_last_name: "Last name #{i}")
+        ect = create(:ect_at_school_period, teacher:, school:, started_on:, finished_on: nil)
+        create(:mentorship_period, mentor:, mentee: ect, started_on:, finished_on: nil)
+        teacher
+      end
+    end
+
+    it 'lists ECT names' do
+      render_inline(described_class.new(mentor:, school:))
+
+      ects.each do |teacher|
+        expect(rendered_content).to include("#{teacher.trs_first_name} #{teacher.trs_last_name}")
```

Comment:
I've always preferred the string without interpolation. I think I'm in the minority

Reply:
I think you might be but that's ok 😄  

I used interpolation here because: 

  - We are defining the factories above
  - We keep things DRY
  - It prevents unecessary coupling and allows the test to be slightly more flexible

Reply:
Let me try and win you over.

- factories are still needed, no change there
- it's repeating code from a service
- it's no less coupled but is less readable

Reply:
I think testing exact strings is valuable too (and I've done that on lines 64/65) but I reckon there is a limit. 

Like, if I had to generate 500 different items, would it still make sense to hardcode 500 expectations?

At that point, interpolation feels like a better trade off, it keeps the test flexible and avoids too much repetition but still checks that the output matches the data.

Reply:
I miss read, thought it was just one teacher you were asserting not all. I concede.